src/typings/git.d.ts

```diff
@@ -174,17 +186,34 @@ export interface Repository {
 
 	blame(path: string): Promise<string>;
 	log(options?: LogOptions): Promise<Commit[]>;
+
+	commit(message: string, opts?: CommitOptions): Promise<void>;
+}
+
+export interface RemoteSource {
+	readonly name: string;
+	readonly url: string;
+}
+
+export interface RemoteSourceProvider {
+	readonly name: string;
+	readonly supportsQuery?: boolean;
+	getRemoteSources(query?: string): ProviderResult<RemoteSource[]>;
 }
 
 export type APIState = 'uninitialized' | 'initialized';
 
-export interface API {
+export interface GitAPI {
```

Comment:
I've also renamed `API` to `GitAPI`. I should probably have done this to begin with in the Git extension, but it's too late now. Remember to always rename it, when getting the latest API from the git extension.